File: Gemini/C++/2-3.c
#include <iostream>
#include <fstream>

using namespace std;

int main() {
    ifstream inputFile("input.txt");
    ofstream outputFile("output.txt");

    if (!inputFile.is_open()) {
        cout << "Error opening input file." << endl;
        return 1;
    }

    if (!outputFile.is_open()) {
        cout << "Error opening output file." << endl;
        return 1;
    }

    char ch;
    while (inputFile.get(ch)) {
        outputFile << ch;
    }

    inputFile.close();
    outputFile.close();

    cout << "File copied successfully." << endl;

    return 0;
}

File: Gemini/Python/2-3.py
def copy_file(input_file, output_file):
    try:
        with open(input_file, 'r') as f_in, open(output_file, 'w') as f_out:
            for line in f_in:
                f_out.write(line)
        print("File copied successfully.")
    except FileNotFoundError:
        print("Error: Input file not found.")

input_file = input("Enter the input file name: ")
output_file = input("Enter the output file name: ")
copy_file(input_file, output_file)

